### PR TITLE
Add appendix for effort, highlighting the key features of our applica…

### PR DESCRIPTION
**Pull request by zhengyup**
Not in ss : Notes functionality 

Screenshot 2023-11-11 at 2 25 53 AM

for #196

### CODE REVIEW

File: docs/_markbind/layouts/default.md [hunk 1]
```diff
@@ -45,6 +45,7 @@
   * [Documentation, logging, testing, configuration, dev-ops]({{ baseUrl }}/DeveloperGuide.html#documentation-logging-testing-configuration-dev-ops)
   * [Appendix: Requirements]({{ baseUrl }}/DeveloperGuide.html#appendix-requirements)
   * [Appendix: Instructions for manual testing]({{ baseUrl }}/DeveloperGuide.html#appendix-instructions-for-manual-testing)
+  * [Appendix: Effort]({{ baseUrl }}/DeveloperGuide.html#appendix-effort)
```
Comment (tim-pipi) [inline docs/_markbind/layouts/default.md:48]: Wah! Good find here man!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1185,3 +1185,41 @@ Expected: The most recent window size and location is retained.
...
+additional functionalities for detailed application viewing and adding of notes that demand larger screen 
+real estate. In our commitment to user experience, we also interactivity with the GUI through an ability 
+to view applications from the list view by clicking on the corresponding card.
+
+### Designing Notes Functionality
+Notes play an integral part in LetsGetHired, allowing users to add many pieces of information associated 
+with an internship application. We deliberated on how notes should be created, viewed, deleted and integrated 
+into LetsGetHired's existing and future functionalities, such as searching for an application by notes. 
+This addition allowed us to introduce another dimension to LetsGetHired, adding depth and nuance to the 
+application management experience
```
Comment (tim-pipi) [inline docs/DeveloperGuide.md:1223]:
```suggestion
application management experience.
```
Missing a full-stop here!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1185,3 +1185,41 @@ Expected: The most recent window size and location is retained.
...
+internship application organization through features like sort and find. We also provided users the 
+flexibility to attach helpful information to entries, through the introduction of notes.
+
+### Designing Organisation Features
+Designing organizational features in LetsGetHired, such as the Find and Sort commands, presented a notable 
+advancement beyond the capabilities of AB3. While AB3 exclusively features a basic Find function that 
+searches for entries based on matching keywords to names in the address book, in LetsGetHired, the Find 
+command empowers users to conduct searches across various fields associated with internship applications.
+Additionally, the Find command in LetsGetHired accommodates searches across 
+multiple fields simultaneously, enhancing user flexibility.
```
Comment (tim-pipi) [inline docs/DeveloperGuide.md:1203]: Perhaps we could highlight that our `find` command also works with partial match - I think this is a powerful (non-trivial) feature that wasn't easy to implement!

**Review by tim-pipi — CHANGES_REQUESTED** [review]
LGTM! 👍 Just some nits to pick!

**Comment by zhengyup** [comment]
Good catches! @tim-pipi pls have a look and feel free to merge :)

**Review by tim-pipi — APPROVED** [review]
LGTM! Thanks for the changes!

**Pull request merged**